src/webgpu/api/operation/sampling/derivative.spec.ts

```diff
@@ -0,0 +1,216 @@
+    );
+    this.queue.submit([commandEncoder.finish()]);
+
+    return buffer;
+  }
+
+  private pipeline: GPURenderPipeline | undefined;
+  async init(): Promise<void> {
+    await super.init();
+
+    this.pipeline = this.device.createRenderPipeline({
+      vertex: {
+        module: this.device.createShaderModule({
+          code: `
+            struct Outputs {
+              [[builtin(position)]] Position : vec4<f32>;
+              [[location(0)]] fragUV : vec2<f32>;
+            };
+
+            [[stage(vertex)]] fn main(
+              [[builtin(vertex_index)]] VertexIndex : u32) -> Outputs {
+              // Full screen quad
+              var position : array<vec3<f32>, 6> = array<vec3<f32>, 6>(
+                vec3<f32>(-0.5, 0.5, 0.0),
+                vec3<f32>(0.5, 0.5,  0.0),
+                vec3<f32>(0.5, -0.5, 0.0),
+                vec3<f32>(0.5, -0.5, 0.0),
+                vec3<f32>(-0.5, -0.5,  0.0),
+                vec3<f32>(-0.5, 0.5,  0.0));
+
+              // Map UVs so that dpdx and dpdy return specific values for the first fragment quad
+              let umax = 0.1; // dx
+              let vmax = 0.2; // dy
+              var uv : array<vec2<f32>, 6> = array<vec2<f32>, 6>(
+                vec2<f32>(0.0, 0.0),
+                vec2<f32>(umax, 0.0),
+                vec2<f32>(umax, vmax),
+                vec2<f32>(umax, vmax),
+                vec2<f32>(0.0, vmax),
+                vec2<f32>(0.0, 0.0));
+
+              var output : Outputs;
+              output.fragUV = uv[VertexIndex];
+              output.Position = vec4<f32>(position[VertexIndex], 1.0);
+              return output;
+            }
+            `,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: this.device.createShaderModule({
+          code: `
+            [[block]] struct Uniforms {
+              numIterations : i32;
+            };
+            [[binding(0), group(0)]] var<uniform> uniforms : Uniforms;
+
+            [[stage(fragment)]] fn main(
+              [[builtin(position)]] FragCoord : vec4<f32>,
+              [[location(0)]] fragUV: vec2<f32>) -> [[location(0)]] vec4<f32> {
+                
+                // Loop to exercise uniform control flow of gradient operations, to trip FXC's
+                // warning X3570: gradient instruction used in a loop with varying iteration, attempting to unroll the loop
+                var r : f32 = 0.0;
+                for (var i = 0; i < uniforms.numIterations; i = i + 1) {
+                  
+                  // Bogus condition to make this a "loop with varying iteration".
+                  if (fragUV.x > 500.0) {
+                    break;
+                  }
+                  
+                  // Do the gradient operations within the loop
+                  let dx = dpdxCoarse(fragUV.x);
+                  let dy = dpdyCoarse(fragUV.y);
+                    r = r + dx + dy;
+                }
+                return vec4<f32>(r);
```

Comment:
nit: You could easily return `vec4<f32>(summed_dx, summed_dy, 0.0, 1.0)` for slightly more thorougher testing.

Reply:
Great idea. Done.